TombEngine/Game/camera.cpp

```diff
@@ -323,6 +332,131 @@ void MoveCamera(GameVector* ideal, int speed)
+	
+	UpdateCameraElevation();
+	
+	int moveSpeed = 1;
+	//get mesh 0 coordinates
+	auto pos = Vector3Int::Zero;
+	GetJointAbsPosition(CamSlotId, &pos, 0);
+	auto dest = Vector3(pos.x, pos.y, pos.z);
+
+	from = GameVector(dest, CamSlotId->RoomNumber);
+	Camera.fixedCamera = true;
+
+	MoveObjCamera(&from, CamSlotId, CamMeshID, TargetItem, TargetMeshID);
+			Camera.timer = -1;
+}
+
+
+void MoveObjCamera(GameVector* ideal, ItemInfo* CamSlotId, int CamMeshID, ItemInfo* TargetItem, int TargetMeshID)
+ {
+	GameVector from, to;
+		
+	int	speed = 1;
+		//Get mesh1 to attach camera to
+	auto pos = Vector3Int::Zero;
+	GetJointAbsPosition(CamSlotId, &pos, CamMeshID);
+		//Get mesh2 to attach target to
+	auto pos2 = Vector3Int::Zero;
+	GetJointAbsPosition(TargetItem, &pos2, TargetMeshID);
+
+	if (
+		OldCam.pos.Position.x != pos.x ||
+		OldCam.pos.Position.y != pos.y ||
+		OldCam.pos.Position.z != pos.z ||
+		OldCam.targetDistance != Camera.targetDistance ||
+		OldCam.targetElevation != Camera.targetElevation ||
+		OldCam.actualElevation != Camera.actualElevation ||
+		OldCam.actualAngle != Camera.actualAngle ||
+		OldCam.target.x != Camera.target.x ||
+		OldCam.target.y != Camera.target.y ||
+		OldCam.target.z != Camera.target.z ||
+		Camera.oldType != Camera.type ||
+		BinocularOn)
+	{
+		OldCam.pos.Position.x = pos.x;
+		OldCam.pos.Position.y = pos.y;
+		OldCam.pos.Position.z = pos.z;
+		OldCam.targetDistance = Camera.targetDistance;
+		OldCam.targetElevation = Camera.targetElevation;
+		OldCam.actualElevation = Camera.actualElevation;
+		OldCam.actualAngle = Camera.actualAngle;
+		OldCam.target.x = Camera.target.x;
+		OldCam.target.y = Camera.target.y;
+		OldCam.target.z = Camera.target.z;
+		LastIdeal.x = pos.x;
+		LastIdeal.y = pos.y;
+		LastIdeal.z = pos.z;
+		LastIdeal.roomNumber = ideal->roomNumber;
+		LastTarget.x = pos2.x;
+		LastTarget.y = pos2.y;
+		LastTarget.z = pos2.z;
+			 }
+	else
+	{
+		pos.x = LastIdeal.x;
+		pos.y = LastIdeal.y;
+		pos.z = LastIdeal.z;
+		ideal->roomNumber = LastIdeal.roomNumber;
+		pos2.x = LastTarget.x;
+		pos2.y = LastTarget.y;
+		pos2.z = LastTarget.z;
+	}
+
+	Camera.pos.x += (ideal->x - Camera.pos.x) / speed;
+	Camera.pos.y += (ideal->y - Camera.pos.y) / speed;
+	Camera.pos.z += (ideal->z - Camera.pos.z) / speed;
+
+	Camera.pos.roomNumber = GetCollision(Camera.pos.x, Camera.pos.y, Camera.pos.z, Camera.pos.roomNumber).RoomNumber;
+	LookAt(&Camera, 0);
+
+	// write last frame camera angle to LastAngle to compare if next frame camera angle has a bigger step than 100.
+	// To make camera movement smoother a speed of 2 is used. While for big camera angle steps (cuts) -
+	// the speed is set to 1 to make the cut immediatelly.
+	int anglex = Camera.target.x - Camera.pos.x;
+	int angley = Camera.target.y - Camera.pos.y;
+	int anglez = Camera.target.z - Camera.pos.z;
+
+	if (LastAnglex - anglex > 100 || 
+		LastAngley - angley > 100 ||
+		LastAnglez - anglez > 100 
+		)
+		{
+		speed = 1;
+		}
+	else
+		{
```

Comment:
TEN has a bit of a funny convention: where an `if` condition takes up one line _and_ the contents of its block also take up one line, we can leave out the brackets.